### PR TITLE
Better waiting for nfs test server.

### PR DESCRIPTION
**Pull request by andreas-kupries**
Ref https://trello.com/c/UL1n6gO0/1007-nfs-persi-brain-failing-on-nightly-caasp3
Additional inspection on exit/failure.

## Description

Help debugging brain test issues.

## Test plan

Run brain tests on various platforms.

### CODE REVIEW

File: src/scf-release/src/acceptance-tests-brain/test-scripts/011_nfspersi_test.rb [hunk 1]
```diff
@@ -33,6 +34,15 @@
         run "cf delete-service -f #{VOLUME_NAME}"
         run "cf disable-service-access persi-nfs"
 
+        # Get logs and more information about the test server, why it
```
Comment (f0rmiga) [inline src/scf-release/src/acceptance-tests-brain/test-scripts/011_nfspersi_test.rb]: I don't think we should merge with these changes. Maybe mark the PR as WIP and have it to run on the various CIs from the spitted images and helm charts from this PR?

File: src/scf-release/src/acceptance-tests-brain/test-scripts/011_nfspersi_test.rb [hunk 2]
```diff
@@ -50,7 +60,15 @@
...
 run "kubectl create -n #{NS} -f #{SKUBEC}"
-wait_for_namespace NS
+
+
+# Do not run the wait below into the full timeout for the test
+# case. We wish to do some inspection and proper cleanup when
+# we abort.  So we wrap a timeout so that we get two minutes to
+# do any cleanup we need.
+Timeout::timeout(ENV.fetch('TESTBRAIN_TIMEOUT', '600').to_i - 120) do
+  wait_for_statefulset(NS, "nfs-test-server")
```
Comment (f0rmiga) [inline src/scf-release/src/acceptance-tests-brain/test-scripts/011_nfspersi_test.rb]: We've been using 4-space indentation for Ruby.

File: src/scf-release/src/acceptance-tests-brain/test-scripts/testutils.rb [hunk 3]
```diff
@@ -185,6 +185,13 @@ def cf_curl(*args)
     JSON.parse output
 end
 
+def wait_for_statefulset(namespace, statefulset, sleep_duration=10)
+  loop do
```
Comment (f0rmiga) [inline src/scf-release/src/acceptance-tests-brain/test-scripts/testutils.rb]: Same here, 4-space indentation.

File: src/scf-release/src/acceptance-tests-brain/test-scripts/011_nfspersi_test.rb (hunk 2 above)
Comment (andreas-kupries) [reply src/scf-release/src/acceptance-tests-brain/test-scripts/011_nfspersi_test.rb]: Will fix. Running on emacs defaults.

Comment (andreas-kupries) [reply src/scf-release/src/acceptance-tests-brain/test-scripts/011_nfspersi_test.rb]: Done with commit 4d305948ad6992258efd08bdeda61d6ee3e81ab1

File: src/scf-release/src/acceptance-tests-brain/test-scripts/testutils.rb (hunk 3 above)
Comment (andreas-kupries) [reply src/scf-release/src/acceptance-tests-brain/test-scripts/testutils.rb]: Done with commit 4d305948ad6992258efd08bdeda61d6ee3e81ab1

**Comment by andreas-kupries** [comment]
Agreed on the better timeout from #2306

File: src/scf-release/src/acceptance-tests-brain/test-scripts/testutils.rb [hunk 4]
```diff
@@ -185,6 +185,13 @@ def cf_curl(*args)
     JSON.parse output
 end
 
+def wait_for_statefulset(namespace, statefulset, sleep_duration=10)
```
Comment (f0rmiga) [inline src/scf-release/src/acceptance-tests-brain/test-scripts/testutils.rb]: How about writing this like https://github.com/SUSE/scf/blob/develop/src/scf-release/src/acceptance-tests-brain/test-scripts/testutils.rb#L160-L168?

```ruby
def wait_for_statefulset_ready(name, namespace, timeout=300)
    run %W(
        kubectl wait statefulset/#{name}
            --for condition=Ready
            --namespace #{namespace}
            --timeout #{timeout}s
    ).join(" ").chomp
end
```

**Comment by andreas-kupries** [comment]
Rebased to head of dev, pulling the new timeout work, replacing what was here.

**Pull request merged**